payments-core/src/main/java/com/stripe/android/payments/paymentlauncher/PaymentLauncherViewModel.kt

```diff
@@ -104,7 +111,20 @@ internal class PaymentLauncherViewModel @Inject constructor(
     ) {
         if (hasStarted) return
         viewModelScope.launch {
-            savedStateHandle.set(KEY_HAS_STARTED, true)
+            savedStateHandle[KEY_HAS_STARTED] = true
+            savedStateHandle[KEY_CONFIRM_ACTION_REQUESTED] = true
+
+            val analyticsParams: Map<String, String> = mapOf(
```

Comment:
can you pull out creating the analyticsParams and logging the analytic via analyticsRequestExecutor into a helper function? Otherwise this makes it a bit harder to easily read through and understand what `confirmStripeIntent` does